### DISCUSSION TITLE
minio on top of 3 node BM k8s

### DISCUSSION THREAD

**Discussion by sivarp** [minio/minio#20484] [Q&A]
I have sized 3 node kubernetes cluster in BM environment with highly available etcd and control plane components. currently, evaluating minio as an option for object storage use case and see that minio recommends 4 node cluster for fault tolerant and HA needs.

Requirement is to tolerate one node down scenario and 2 disks crashed at a time given below design,
server pool - 3
disks per node - 2
parity class - EC: 2

I pulled down plug on a node and minio is able to serve the object as expected with erasure coding doing its job. but then, why node count 4 is recommended wherever I read, could you please enlighten ?

I see in my particular case, 3 node or 4 node works similar when I want to tolerate one node / two disks crash scenario.

**Comment by klauspost** [comment]
There is basically no margin for failure, or resolving inconsistencies. If you only have 3 nodes one out will mean you are on the edge of keeping your data.

For HA discussion reach out on `sales@min.io`.
